File: music-domain/src/main/java/com/music/sale/domain/user/enums/UserRole.java
package com.music.sale.domain.user.enums;
public enum UserRole {
    ADMIN,
    SELLER,
    USER;

    public boolean hasPermission(UserRole requiredRole) {
        return switch (this) {
            case ADMIN -> true;
            case SELLER -> this == requiredRole;
            case USER -> this == requiredRole;
        };
    }
}

File: src/main/kotlin/com/sodamjae/domain/user/enum/UserRole.kt
package com.sodamjae.domain.user.enum

enum class UserRole {
    ADMIN, SELLER, USER;

    fun hasPermission(requiredRole: UserRole): Boolean {
        return when (this) {
            ADMIN -> true
            SELLER -> this == requiredRole
            USER -> this == requiredRole
        }
    }
} 